### ISSUE TITLE
OEM partition snapshotting

### ISSUE THREAD

**Issue by anmazzotti** [rancher/elemental-toolkit#2203]
This is a new feature request for OEM partition snapshotting.  

The desire is to be able to create new snapshots when running `elemental upgrade`.  
It also probably makes sense to bind them to system snapshots somehow, so that they can be rolled back together seamlessly.  

I am also not sure what the behavior for taking the snapshot should be. Shall the toolkit simply take a snapshot of current OEM, or should the user input a path to take a snapshot from?

**Comment by davidcassany** [comment]
Yes, ideall would also be nice to check what Micro does for `/etc`, for which a separate overlayfs layer is kept in persistent area for each snapshot. I wonder if it would make sense adding a flag to _snapshot_ somehow the persistent directories we overlay. Not sure how to handle persistent volumes though 🤔 should we start treating `/oem` as a special case?
